### PR TITLE
feat(img-proxy)

### PR DESCRIPTION
**Pull request by stnw**
https://www.notion.so/hellocharles/WIP-Product-image-s-resizing-d3afe1ae598e4d3a853cbacad93a61ce

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -157,6 +157,7 @@
     "just-typeof": "^2.0.0",
     "mqtt": "^4.2.6",
     "qs": "^6.6.0",
+    "query-string": "^7.0.1",
```
Comment (eljefedelrodeodeljefe) [inline package.json:160]: please remove this dependency, as the `qs` is doing exactly the same. Was there any reason for it? In any case we prefer `qs`, as it support everything the API suipports

File: src/entities/image-proxy/image-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      return response.headers.code === 200
+    } catch (error) {
+      throw new UrlShortenerImgProxyError({ error })
+    }
+  }
+
+  public getResizedImgUrl (request: ImageProxyResizeRequest): string {
+    if (!this.configuration) throw new TypeError('img proxy requires configuration to be set.')
+
+    return `${this.configuration.domain}/proxy/images?${queryString.stringify(this.getUrlParams(request))}`
```
Comment (eljefedelrodeodeljefe) [inline src/entities/image-proxy/image-proxy.ts:163]: see note above

File: src/entities/image-proxy/image-proxy.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      request.format === null ? null : { format: request.format },
+      request.background === null ? null : { background: request.background },
+      request.gravity === null ? null : { gravity: request.gravity }
+    )
+
+    if (request.resize) {
+      params['resize[type]'] = request.resize.type
+      params['resize[height]'] = request.resize.height
+      params['resize[width]'] = request.resize.width
+    }
```
Comment (eljefedelrodeodeljefe) [inline src/entities/image-proxy/image-proxy.ts:178]: can't you not just use normal object notation and use `qs` for the stringified version?

File: src/universe/index.ts [hunk 4]
```diff
@@ -2100,6 +2100,10 @@ export class Universe extends APICarrier {
     return await this.makeBaseResourceListRequest<urlShortener.UrlShortener, urlShortener.UrlShorteners, urlShortener.UrlShortenerRawPayload, EntityFetchOptions, urlShortener.UrlShortenersFetchRemoteError>(urlShortener.UrlShortener, urlShortener.UrlShorteners, urlShortener.UrlShortenersFetchRemoteError, options)
   }
 
+  public async imgProxies (options?: EntityFetchOptions): Promise<imgProxy.ImageProxy[] | imgProxy.ImageProxyRawPayload[] | undefined> {
```
Comment (eljefedelrodeodeljefe) [inline src/universe/index.ts:2103]: would prefer `s/imgProxies/imageProxies`

File: src/entities/image-proxy/image-proxy.ts [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        },
+        data: { },
+        responseType: 'json'
+      }
+
+      const response = await this.http?.getClient()(opts)
+
+      return response.headers.code === 200
+    } catch (error) {
+      throw new UrlShortenerImgProxyError({ error })
```
Comment (eljefedelrodeodeljefe) [inline src/entities/image-proxy/image-proxy.ts:156]: probably missed the move to ImageProxy, no?

**Review by eljefedelrodeodeljefe — CHANGES_REQUESTED** [review]
Generally LGTM, but with a couple of comments to address, especially the `qs` bit

**Review by MaiLinhGroup — APPROVED** [review]
LGTM 🐻

**Pull request merged**